### DISCUSSION TITLE
Understanding the Saral Reference App (School Assessment) flow with APIs used at each step

### DISCUSSION THREAD

**Discussion by surendrasinghs** [Sunbird-Saral/Community#5] [General]
Can you please review if the APIs mapped to the workflow in the app are correct?
I have referred to the OMR layout scanning workflow shown in https://youtu.be/VuYBvj8iN9A and tried mapping it to mandatory APIs required by the frontend as mentioned in the [v1.5.3 release notes](https://github.com/Sunbird-Saral/Project-Saral/releases/tag/v1.5.3)

<img width="1372" alt="image" src="https://user-images.githubusercontent.com/10580754/174049497-a61ba81e-caf4-44a7-9298-93e2c7190e50.png">

Please share if there are any corrections or suggestions.

**Comment by pramodkvarma** [comment]
We should clearly separate the reference app flow from the core SARAL SDK. Same SDK can be used in Agri or other domains for example and reference app is only built for school assessment. So, thinking everything from reference app along is very limiting. @surendrasinghs @kumardeepak

**Reply by surendrasinghs** [reply]
Yes, I understand and agree... I have renamed the discussion thread to say "Saral Reference App (School Assessment)" flow. Thanks for highlighting this.

I will contribute this to the microsite as well so that it reads "Saral Reference App for School Assessment" wherever Saral App is mentioned. Thanks!

**Comment by dileep-gadiraju** [comment]
@surendrasinghs,

also here is the feedback on how the APIs are used within the workflow.

1. `GET  /brand/default` is used before the login process but `GET  /brand` is called after successful login through POST /schools/login API.
2. After Reviewing and correcting the scanned data by the user, the flow will lead to saving results in phone local storage. `Skip` comes into the picture only for multi-student layouts.  `PUT  /saveMarks` is called in the last step "Back to Exam Details. >> Save All Scan". That's when data gets pushed to the backend from local storage.
3. Also FYI there is [auto-sync feature](https://saral.sunbird.org/learn/features/auto-sync) which also tries to push data from local storage to the backend based on the configuration. Refer to the documentation link for more details. 

Hope this helps.

**Reply by surendrasinghs** [reply]
Thanks, @dileep-gadiraju. Please see if the updated diagram below looks okay.

<img width="1190" alt="image" src="https://user-images.githubusercontent.com/10580754/174303723-6d83da90-c0d0-4cc2-beaa-63d483a8dcfb.png">

I would be happy to contribute this as an explanation of the _Saral Reference App for School Assessment_ on the documentation microsite.

**Reply by pramodkvarma** [reply]
@surendrasinghs Why don't you do a PR for Saral ref app documentation and include this. @dileep-gadiraju can review and accept the PR.

**Reply by dileep-gadiraju** [reply]
Sure @pramodkvarma.
